xivlauncher-rb: init at 1.0.8.2

drakon64 opened the pull request:
## Description of changes

Package `xivlauncher-rb`, a fork of the existing `xivlauncher` package. This fork adds the option to use newer Wine builds than the standard XIVLauncher provides (including Wayland-enabled builds) and also newer DXVK builds.

As this is a fork of an existing package, the new package is built by overriding the existing `xivlauncher` package with `useRbPatchedLauncher = true;`

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

drakon64 committed:
xivlauncher-rb: init at 1.0.8.2

keysmashes reviewed (commented):
I have not tested the -rb launcher, and cannot commit to doing so in the future, but I don't have any particularly compelling argument as to why this shouldn't be merged if someone else can.

It would be nice to have `xivlauncher` and `xivlauncher-rb`, rather than `xivlauncher.override { useRbPatchedLauncher = true; }`; unfortunately I don't think there's a particularly great way of doing that with by-name packages right now. (maybe you could add something to all-packages.nix, but then you'd end up duplicating an awful lot of the `postFixup` machinery, and I don't know how it'd interact with `buildDotnetModule`.) I had a brief go at getting `xivlauncher` and `xivlauncher.rb` to work (via `passthru.rb`) but couldn't figure it out.

The other alternative I guess would be to do what nvtop did, and have `xivlauncherPackages.default` and `xivlauncherPackages.rb`, but that seems strictly worse...

drakon64 committed:
xivlauncher-rb: Avoid adding additional uppercase letters to `pname`

Co-authored-by: ash <ash@sorrel.sh>

drakon64 committed:
xivlauncher-rb: Inherit `owner`, `rev`, and `hash` from `source`

Co-authored-by: ash <ash@sorrel.sh>

drakon64 committed:
xivlauncher-rb: Additional newline to reduce future diff sizes

Co-authored-by: ash <ash@sorrel.sh>

drakon64 committed:
xivlauncher-rb: Add `zstd` to `steam-run` environment
Most of the Wine versions available within the launcher now use zstd compression rather than xz. This is required to resolve https://github.com/rankynbass/XIVLauncher.Core/issues/17.

drakon64 commented on pkgs/by-name/xi/xivlauncher/package.nix:
```diff
@@ -41,7 +58,7 @@ in
     postFixup = lib.optionalString useSteamRun (let
       steam-run = (steam.override {
-        extraPkgs = pkgs: [ pkgs.libunwind ];
+        extraPkgs = pkgs: [ pkgs.libunwind ] ++ lib.optionals useGameMode [ pkgs.zstd pkgs.gamemode ];
```
Is there a better way to add `zstd` support?

drakon64 commented on pkgs/by-name/xi/xivlauncher/package.nix:
```diff
@@ -41,7 +58,7 @@ in
 
     postFixup = lib.optionalString useSteamRun (let
       steam-run = (steam.override {
-        extraPkgs = pkgs: [ pkgs.libunwind ];
+        extraPkgs = pkgs: [ pkgs.libunwind ] ++ lib.optional useGameMode pkgs.gamemode ++ lib.optional useRbPatchedLauncher pkgs.zstd;
```
Is there a better way to add zstd support?
